sync/clients/sync.py

```diff
@@ -1,14 +1,48 @@
+import difflib
+import json
 import logging
-from typing import Generator
+from typing import Dict, Generator
 
 import httpx
+from pydantic import BaseModel
 
 from ..config import API_KEY, CONFIG, APIKey
 from . import USER_AGENT, RetryableHTTPClient, encode_json
 
 logger = logging.getLogger(__name__)
 
 
+class S3ClusterLogConfiguration(BaseModel):
+    destination: str
+    region: str
+    enable_encryption: bool
+    canned_acl: str
+
+
+class DBFSClusterLogConfiguration(BaseModel):
+    destination: str
+
+
+class ClusterLogConfiguration(BaseModel):
+    s3: S3ClusterLogConfiguration
+    dbfs: DBFSClusterLogConfiguration
+
+
+class ProjectConfiguration(BaseModel):
+    node_type_id: str
+    driver_node_type: str
+    custom_tags: Dict
+    cluster_log_conf: ClusterLogConfiguration
```

Comment:
Should this be a Union[S3ClusterLogConfiguration, DBFSClusterLogConfiguration] ? it can be either or but can't be both or neither